### DISCUSSION TITLE
Expand does not seem to work on the js sdk

### DISCUSSION THREAD

**Discussion by Mitschiro** [pocketbase/pocketbase#5794] [Q&A] [answered]
JS SDK: v0.21.5
Pocketbase: v0.22.21
Schemas: [pb_schema.json](https://github.com/user-attachments/files/17614789/pb_schema.json)

Code:

![image](https://github.com/user-attachments/assets/b6757a0d-7e8e-4ea2-8346-cdd6aa03c365)

I've tried it several times and it never works, even if i leave the view and list/search rules empty so everyone has access.

The only time this works is if i use the admin account, which is not an option in a client app.

And for context, the auth user doing the request here is within the rule boundaries.

Am i missing something?

Thanks in advance for the answers.

**Comment by ganigeorgiev** [comment, marked as answer]
_Please avoid posting code as screenshot._

The expanded data will be under the `expand` property and you are excluding that field with the custom `fields` list. Either remove `fields` or add `expand` to its list.

~If you are still not able to make it work, please export your collections configuration as json (Admin UI > Settings > Export collections).~ On second read, I just noticed the link to your schema. I've tested it locally it works correctly for me as long as the `fields` list is properly defined and your client request satisfies the View API rule of the `periods` collection.

**Reply by ganigeorgiev** [reply]
_Also if you are using Next.js make sure that what you are experiencing is not caused by its default cache._

**Reply by Mitschiro** [reply]
This worked now, thank you for your quick response.

I see, the fields option invalids the expand option.

But shouldn't that in theory not happen if the field is listed in both options though?
If you request it in fields, you should be able to expand on it if the rules apply ofc.

Not that problematic right now as i can handle the fields part via code but as you scale this might become a pain point.

I've not yet looked into indexes and views, pls don't lynch me.

Have a great day.

**Reply by Mitschiro** [reply]
> _Also if you are using Next.js make sure that what you are experiencing is not caused by its default cache._

I'm using [nuxt](https://nuxt.com/).
This code is executed on the server, so no caching but thanks for the heads-up :)

**Reply by ganigeorgiev** [reply]
>  But shouldn't that in theory not happen if the field is listed in both options though?

That's the problem - it is NOT listed in the `fields`. When you expand data, the expanded data is added under a new `expand.periods` property. The original relation field is left unchanged (which is either a single id or array of ids). Console log the full json to help you understand better what is going on.

**Reply by Mitschiro** [reply]
> > But shouldn't that in theory not happen if the field is listed in both options though?
> 
> That's the problem - it is NOT listed in the `fields`. When you expand data, the expanded data is added under a new `expand.periods` property. The original relation field is left unchanged (which is either a single id or array of ids). Console log the full json to help you understand better what is going on.

OOHHH, fml.

Yeah, makes sense if you think about it more than 10 seconds, which i clearly did not.
I thought the `fields` option only affects the record model as `expand` is not an original member of it and not the entire response model.
Adding the `expand` field into the `fields` option returns it correctly now.

Thank you.
